platform/mbed_wait_api_rtos.cpp

```diff
@@ -31,23 +31,39 @@ void wait(float s)
 
 void wait_ms(int ms)
 {
-    wait_us(ms * 1000);
+    // Wait equivalent to 1 ms near the tick might result into zero wait.
+    // Hence ms precision can be used for > 1 ms wait only.
+    if ((ms >= 1) && core_util_are_interrupts_enabled()) {
```

Comment:
`wait_us` already chooses between thread wait and busy wait. Why do you need the checking here?

Reply:
ThisThread::sleep_for(ms); in the if condition will assert if inside the interrupt.

Reply:
`ms >= 1`? If it's not, it's negative or zero.

And I would say "wait_ms()" if interrupts are disabled is breaking the system. Interrupts should not be disabled for as long as a millisecond, so I don't like coding to permit this. Although I guess the existing code does do it, so maybe backwards compatibility. But if changing anyway... 

People willing to do a big wait in interrupt could do it as `wait_us(5000)` where the big number kind of exposes the enormity of how bad they're being.